### PR TITLE
create_dir_winapi: update winapi cmd

### PR DESCRIPTION
**Pull request by xiagao**
To compatible with python3, winapi cmd should be
updated.
ID: 2239

### CODE REVIEW

**Comment by xiagao** [comment]
@hellohellenmao Hi, call for your review,thanks.

**Comment by xiagao** [comment]
(1/1) Host_RHEL.m9.u5.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.Win2025.x86_64.io-github-autotest-qemu.virtio_fs_share_data.with_cache.auto.create_dir_winapi.q35: STARTED
 (1/1) Host_RHEL.m9.u5.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.Win2025.x86_64.io-github-autotest-qemu.virtio_fs_share_data.with_cache.auto.create_dir_winapi.q35: PASS (142.56 s)

File: qemu/tests/cfg/virtio_fs_share_data.cfg [hunk 1]
```diff
@@ -95,7 +95,7 @@
             no Host_RHEL.m8
             fs_binary_extra_options += " --log-level debug"
             winapi_dir_name = test_winapi
-            create_dir_winapi_cmd = python -c "import ctypes; dll=ctypes.windll.kernel32;dll.CreateDirectoryA('%s\\${winapi_dir_name}', 0)"
+            create_dir_winapi_cmd = python -c "import ctypes; kernel32 = ctypes.WinDLL('kernel32', use_last_error=True); kernel32.CreateDirectoryW('%s\\${winapi_dir_name}', 0)"
```
Comment (hellohellenmao) [inline qemu/tests/cfg/virtio_fs_share_data.cfg:98]: What about the old pattern, do we still need to consider the compatilbility of the python version lower than python3? I mean that here we can not delete the old patten directly, but to make it compatible for both old and current python, right?

**Review by hellohellenmao — CHANGES_REQUESTED** [review]

File: qemu/tests/cfg/virtio_fs_share_data.cfg (hunk 1 above)
Comment (xiagao) [reply qemu/tests/cfg/virtio_fs_share_data.cfg:98]: The new cmd is compatible for both.

Comment (hellohellenmao) [reply qemu/tests/cfg/virtio_fs_share_data.cfg:98]: Ah get it. Thanks.

**Comment by hellohellenmao** [comment]
With the PR, the case passed:
 (2/2) Host_RHEL.m10.u0.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.Win2022.x86_64.io-github-autotest-qemu.virtio_fs_share_data.with_cache.auto.create_dir_winapi.q35: PASS (439.94 s)
Without, the case failed:
(2/2) Host_RHEL.m10.u0.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.Win2022.x86_64.io-github-autotest-qemu.virtio_fs_share_data.with_cache.auto.create_dir_winapi.q35: FAIL: Create dir failed in 60s, output is  Vo***

**Review by hellohellenmao — APPROVED** [review]
LGTM

**Comment by xiagao** [comment]
@YongxueHong Could you help review this patch? Thanks.

**Review by YongxueHong — APPROVED** [review]
LGTM.

**Comment by YongxueHong** [comment]
Hi @xiagao 
There is a prompt: `This branch is out-of-date with the base branch`, pls rebase your develop branch to the latest.
Thanks.

**Comment by xiagao** [comment]
> Hi @xiagao There is a prompt: `This branch is out-of-date with the base branch`, pls rebase your develop branch to the latest. Thanks.

Updated @YongxueHong ,thanks reminder.

**Comment by xiagao** [comment]
@YongxueHong Update with GPG signature. Could you take a review again?

**Pull request merged**